### PR TITLE
Disable Agg push down for incremental scan

### PR DESCRIPTION
**Pull request by huaxingao**
Disable Agg push down for incremental scan for now. Will have a followup to enable Agg push down for incremental scan.

### CODE REVIEW

**Review by amogh-jahagirdar — APPROVED** [review]
Looks good to me with just a nit, I'll hold for other's feedback, cc: @RussellSpitzer @aokolnychyi @singhpk234

File: spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java [hunk 1]
```diff
@@ -222,12 +222,19 @@ public boolean pushAggregation(Aggregation aggregation) {
     }
 
     TableScan scan = table.newScan().includeColumnStats();
-    Snapshot snapshot = readSnapshot();
-    if (snapshot == null) {
-      LOG.info("Skipping aggregate pushdown: table snapshot is null");
+    if (readConf.startSnapshotId() == null) {
```
[the remark this answers is not in the document]
Comment (huaxingao) [reply spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java]: Sounds good. Fixed.

**Review by nastra — APPROVED** [review]

**Review by singhpk234 — APPROVED** [review]
LGTM as well, Thanks @huaxingao

**Review by ChristinaTech — APPROVED** [review]

**Review by aokolnychyi — APPROVED** [review]

**Comment by amogh-jahagirdar** [comment]
Thanks @huaxingao for the quick fix!

**Pull request merged**